test/SILGen/cf_members.swift

```diff
@@ -15,7 +15,7 @@ public func importAsUnaryInit() {
 // CHECK-LABEL: sil [ossa] @$s10cf_members3foo{{[_0-9a-zA-Z]*}}F
 public func foo(_ x: Double) {
 // CHECK: bb0([[X:%.*]] : $Double):
-  // CHECK: [[GLOBALVAR:%.*]] = global_addr @IAMStruct1GlobalVar
+  // CHECK: [[GLOBALVAR:%.*]] = global_addr @_Z19IAMStruct1GlobalVar
```

Comment:
Even though this change has no effect on IRGen (I think), mangling these specific names in SIL is weird because they are not actually mangled in the object file.

Reply:
Changed the ASTMangler to use Clang mangler only for var decls that don't come from the top level context. I.e. we're not mangling var decls from translation unit, but we do mangle var decls from records or namespaces.